engine/random.cpp

```diff
@@ -0,0 +1,42 @@
+#include "random.h"
+
+namespace pong
+{
+
+static std::random_device rd;
+static std::mt19937 generator(rd());
+
+template<typename T>
+T Random::Range(T min, T max)
+{
+    static_assert(false);
+}
+
+template<>
+int Random::Range<int>(int min, int max)
```

Comment:
naming nit: possibly 

```
randomGenerator.GetValueInRange()
```

a method called Range makes me think it will return the range